draft-ietf-quic-transport.md

```diff
@@ -1131,6 +1131,13 @@ client MUST terminate with a QUIC_VERSION_NEGOTIATION_MISMATCH error code if
 version negotiation occurred but it would have selected a different version
 based on the value of the supported_versions list.
 
+The position and size of the version validation fields in the transport
+parameters MUST NOT change between different versions of QUIC.  If a version of
+QUIC needs to define a new format for transport parameters, it MUST define and
+use a new TLS extension rather than redefine the layout of the existing
```

Comment:
@ianswett, let's say that we don't change the format of the transport parameters for version 1.  We also have a server that supports version 1 and some hypothetical version 4.

With the current format, a server that selects version 1 will send advertise transport parameters starting with the bytes 0x0008, 0x00000004, 0x00000001 (a length, followed by version 4 and version 1).  After that, it will have some lengths and TLVs for parameters.

However, if version 4 decides to change the transport parameters, then the client that is subject to spoofing of a version negotiation packet might find that they interpret transport parameters differently.  If version 4 is sufficiently similar to version 1 in other ways, you could even end up in a situation where the client and server disagree on what version they are speaking.

For instance, the above value could be interpreted as being valid against a different schema.